ast/Trees.h

```diff
@@ -247,6 +247,23 @@ class ExpressionPtr {
         using std::swap;
         swap(this->ptr, other.ptr);
     }
+
+    template <typename E> static ExpressionPtr fromUnique(std::unique_ptr<E> ptr) noexcept {
+        if (ptr == nullptr) {
+            return nullptr;
+        }
+        return ExpressionPtr(ExpressionToTag<E>::value, ptr.release());
+    }
+
+    template <typename E> std::unique_ptr<E> toUnique() noexcept {
+        if (this->ptr == 0) {
+            return nullptr;
+        }
+        auto p = as_tree<E>();
```

Comment:
Just to confirm, we're calling `as_tree` instead of using the return value of `release` is so that `bool isValid = tagMaybeZero() == uint16_t(ExpressionToTag<To>::value)` is called, right?